dags/sentinel_2/sentinel-2_gap_filler.py

```diff
@@ -97,13 +101,19 @@ def get_missing_stac_files(s3_report_path, offset=0, limit=None):
     read the gap report
     """
 
+    print(f"Reading the gap report {s3_report_path}")
+
     hook = S3Hook(aws_conn_id=CONN_SENTINEL_2_SYNC)
+    s3 = S3(conn_id=CONN_SENTINEL_2_SYNC)
     bucket_name, key = hook.parse_s3_url(s3_report_path)
-    print(f"Reading the gap report {s3_report_path}")
 
-    files = hook.read_key(key=key, bucket_name=bucket_name).splitlines()
+    missing_scene_file_gzip = s3.get_s3_contents_and_attributes(
+        bucket_name=bucket_name,
+        region=REGION,
+        key=key,
+    )
 
-    for f in files[offset:limit]:
+    for f in gzip.decompress(missing_scene_file_gzip).decode("utf-8").split("\n")[offset:limit]:
```

Comment:
This will do it all in memory, right? I guess it's not a problem, but it might be better to do this row/by/row.

See: https://stackoverflow.com/questions/23091770/how-do-you-iterate-through-a-gzipped-carriage-return-file-using-python-2-7